ui/app/mainui/Handlers/SendModalHandler.qml

```diff
@@ -733,8 +733,24 @@ QtObject {
                 }
 
                 readonly property var totalFeesAggregator: FunctionAggregator {
-                    model: !!handler.fetchedPathModel ?
-                               handler.fetchedPathModel: null
+                    model: {
+                        let noFees = true
+                        for (let i = 0; i < handler.fetchedPathModel.ModelCount.count; i++) {
+                            const item = SQUtils.ModelUtils.get(handler.fetchedPathModel, i)
+                            noFees = noFees && item["fromChainNoBaseFee"] && item["fromChainNoPriorityFee"]
+                            if (!noFees) {
+                                break
+                            }
+                        }
+
+                        if (noFees) {
+                            const nativeTokenSymbol = Utils.getNativeTokenSymbol(simpleSendModal.selectedChainId)
+                            simpleSendModal.estimatedCryptoFees = root.fnFormatCurrencyAmount("0", nativeTokenSymbol)
+                            simpleSendModal.estimatedFiatFees = root.fnFormatCurrencyAmount("0", root.currentCurrency)
```

Comment:
this doesn't seem to be the right place to do this, this part should only handle assign the model to the FunctionAggregator

Reply:
A short and simple explanation is, that in the case of gasless chains aggregate function won't be even triggered and that's why that code is there. That was the first idea that came to my mind.